doc/api/next_api_changes/removals/22107-OG.rst

```diff
@@ -0,0 +1,44 @@
+Removal of deprecated ``mathtext`` APIs
```

Comment:
```suggestion
Removal of deprecated `matplotlib.mathtext` APIs
```
For modules it's usually clearer to state the full name.

Reply:
I seem to have read that it is preferred to not have links in the headings. (But tried to change it elsewhere.)